src/test/java/org/jitsi/videobridge/health/config/HealthTimeoutPropertyTest.java

```diff
@@ -0,0 +1,83 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.health.config;
+
+import com.typesafe.config.*;
+import org.jitsi.testutils.*;
+import org.jitsi.videobridge.util.config.*;
+import org.junit.*;
+
+import static org.junit.Assert.*;
+
+public class HealthTimeoutPropertyTest
+{
+    @Test
+    public void whenOnlyOldConfigProvidesAValue()
+    {
+        Config legacyConfig = ConfigFactory.parseString(HealthConfig.HealthTimeoutProperty.legacyPropName + "=60000");
+        new ConfigSetup()
+            .withLegacyConfig(legacyConfig)
+            .withEmptyNewConfig()
+            .finishSetup();
+
+        HealthConfig.HealthTimeoutProperty healthTimeoutProp = new HealthConfig.HealthTimeoutProperty();
+        assertEquals("The value from the old config should be read correctly", 60000, (int)healthTimeoutProp.get());
+    }
+
+    @Test
+    public void whenOnlyNewConfigIsPresent()
+    {
+        Config newConfig = ConfigFactory.parseString(HealthConfig.HealthTimeoutProperty.propName + "=10 seconds");
+        new ConfigSetup()
+            .withNewConfig(newConfig)
+            .finishSetup();
+
+        HealthConfig.HealthTimeoutProperty healthTimeoutProp = new HealthConfig.HealthTimeoutProperty();
+        assertEquals("The value from the new config should be read correctly", 10000, (int)healthTimeoutProp.get());
+    }
+
+    @Test
+    public void whenOldAndNewConfigsArePresent()
+    {
+        Config legacyConfig = ConfigFactory.parseString(HealthConfig.HealthTimeoutProperty.legacyPropName + "=60000");
+        Config newConfig = ConfigFactory.parseString(HealthConfig.HealthTimeoutProperty.propName + "=10 seconds");
+        new ConfigSetup()
+            .withLegacyConfig(legacyConfig)
+            .withNewConfig(newConfig)
+            .finishSetup();
+
+        HealthConfig.HealthTimeoutProperty healthTimeoutProp = new HealthConfig.HealthTimeoutProperty();
+        assertEquals("The old config value should be used", 60000, (int)healthTimeoutProp.get());
+    }
+
+    @Test
+    public void doesNotChangeAfterConfigReload()
+    {
+        Config newConfig = ConfigFactory.parseString(HealthConfig.HealthTimeoutProperty.propName + "=10 seconds");
+        new ConfigSetup()
+            .withNewConfig(newConfig)
+            .finishSetup();
+
+        HealthConfig.HealthTimeoutProperty healthTimeoutProp = new HealthConfig.HealthTimeoutProperty();
+
+        Config changedConfig = ConfigFactory.parseString(HealthConfig.HealthTimeoutProperty.propName + "=90 seconds");
+        JvbConfig.configSupplier = () -> changedConfig;
+        JvbConfig.reloadConfig();
+
+        assertEquals(10000, (int)healthTimeoutProp.get());
+    }
+}
```

Comment:
I feel like these property tests replicate a lot of the same test code.  Can this be abstracted or templatized in any way?

Reply:
well, i added the helper for test setup, but everything that's left is pretty much what changes from test-to-test.  was there something that specifically stood out?

Reply:
Just that they felt fairly similar, I didn't read them in so much detail.

Reply:
I don't know if it's possible to have a meta-test, but something like "here's the old property name, here's the new property name, here's an appropriate value if you're setting the old property, here's an appropriate value if you're setting the new property -- run through the full set of all the tests."

Reply:
yeah that's an interesting idea...i'll take a shot at that

Reply:
put up something for this...i think it turned out pretty good, should make testing a lot easier.